src/main/java/com/facebook/openwifirrm/modules/Modeler.java

```diff
@@ -340,6 +343,57 @@ private void processData(InputData data) {
 		}
 	}
 
+	/**
+	 * NOTE: if a BSSID does not have a non-obsolete entry, it will be returned
+	 * (i.e., it will not be a key in the returned map).
+	 */
+	public Map<String, WifiScanEntry> getAggregatedWifiScans(long obsoletion_period) {
+		Map<String, WifiScanEntry> aggregatedWifiScans = new HashMap<>();
+		for (String bssid : dataModel.latestWifiScans.keySet()) {
+			// flatten the wifiscan entries and sort in reverse chronological order
+			List<List<WifiScanEntry>> scans = dataModel.latestWifiScans.get(bssid);
+			List<WifiScanEntry> mostRecentToOldest = scans.stream().flatMap(list -> list.stream())
+					.sorted((entry1, entry2) -> {
+						return -Long.signum(entry1.tsf - entry2.tsf);
+					}).collect(Collectors.toUnmodifiableList());
+
+			/*
+			 * For a given BSSID, discard the obsolete entries. Then,consider the most
+			 * recent entry. Take its ht_oper and vht_oper. Discard earlier entries with a
+			 * different ht_oper or a different vht_oper. For earlier entries with the same
+			 * ht_oper and vht_oper, take the average signal value.
+			 */
+			long now = Instant.now().getEpochSecond();
+			String newest_ht_oper = null;
+			String newest_vht_oper = null;
+			double averagedSignal = 0.0;
+			int count = 0;
+			for (WifiScanEntry entry : mostRecentToOldest) {
+				if (now - entry.tsf >= obsoletion_period) {
+					// discard obsolete entries
+					break;
+				}
+				if (newest_ht_oper == null || newest_vht_oper == null) {
+					// start with the most recent entry
+					newest_ht_oper = entry.ht_oper;
+					newest_vht_oper = entry.vht_oper;
+					aggregatedWifiScans.put(bssid, entry);
+					averagedSignal = entry.signal;
+					count++;
+					continue;
+				}
+				if (!entry.ht_oper.equals(newest_ht_oper) || !entry.vht_oper.equals(newest_vht_oper)) {
+					// discard older entries with different ht_oper or different vht_oper
+					continue;
+				}
+				// average signal value from older entries with the same ht_oper and vht_oper
+				averagedSignal = (count / (count + 1)) * averagedSignal + (entry.signal / (count + 1));
```

Comment:
I'd add an input argument for this, not all use cases will want an average signal (say, maybe we want min+max). We can figure this out later as needed.

Also I'm pretty sure this is totally broken because you haven't casted "count" to floating point, so `count / (count + 1)` will always evaluate to 0.